packages/web-components/src/components/ic-date-input/ic-date-input.css

```diff
@@ -50,3 +50,63 @@ ic-input-component-container:hover {
   position: absolute;
   left: -9999px;
 }
+
+.input-container {
+  width: 100%;
+  display: flex;
+  align-items: center;
+  position: relative;
+}
+
+.clear-button,
+.calendar-button {
+  position: absolute;
+  display: block;
+  right: 0;
+  border-radius: var(--ic-border-radius);
+  transition: box-shadow var(--ic-easing-transition),
+    border-radius var(--ic-easing-transition);
+  margin: 0;
+}
+
+.with-calendar {
+  right: 2rem;
+  margin-right: var(--ic-space-xs);
+}
+
+.clear-button.hidden {
+  display: none;
+}
+
+.clear-button:focus,
+.calendar-button:focus {
+  background-color: var(--ic-focus-blue);
+  box-shadow: inset 0 0 0 0.125rem var(--ic-focus-glow);
+  border-radius: 0.25rem;
+}
+
+.clear-button:focus *,
+.calendar-button:focus * {
+  fill: white;
+}
+
+ic-input-component-container.small {
+  --input-width: 16.5rem;
+}
+
+ic-input-component-container.large {
+  --input-width: 22rem;
+}
```

Comment:
It looks to me like it updates the width of the date input as the padding value on the side remains unchanged and gap is always 8px e.g 
<img width="224" alt="Screenshot 2023-08-01 at 07 45 45" src="https://github.com/mi6/ic-ui-kit/assets/111882034/aac9f568-77dd-4a4f-ab3b-082dfee9b4e5">
<img width="224" alt="Screenshot 2023-08-01 at 07 45 51" src="https://github.com/mi6/ic-ui-kit/assets/111882034/aa76f811-406b-4051-8a30-d1c021b62449">
Not sure if I'm looking in the wrong place

Reply:
Sorry, think I've made this confusing. Was trying to work out if the `size` prop should only affect the padding of the input-container rather than increase it's width in addition to the padding. Since we are using `size` as our 'density', density is more to do with how close the input values/textContent is to the border surrounding it rather than the width.

Reply:
Ahh yeah I think this is a weird one because it's meant to match the width of the calendar for date picker which obviously does change width for the different size variants, but then if this was just used on its own would we necessarily want it to change width? Hopefully a designer can weigh in on this

Reply:
Designers agree that it should only affect the padding and the width should only change for date picker to match calendar sizes. This has now been updated.